### DISCUSSION TITLE
How to catch initialization error

### DISCUSSION THREAD

**Discussion by brianheilig** [abacritt/angularx-social-login#416] [Q&A] [answered]
I provided a bad clientId to my GoogleLoginProvider to test that my application handles errors well. It seems I'm unable to catch this error and do anything meaningful with it, such as display a dialog to the user. Here's a bit of code:

```
providers: [
  {
    id: GoogleLoginProvider.PROVIDER_ID,
    provider: new GoogleLoginProvider('a_bad_client_id.apps.googleusercontent.com')
  }
]

```

And in my auth service:

```
constructor(
  private authService: SocialAuthService,
  private http: HttpClient,
  private router: Router) {

  this.authService.initState.subscribe(
    (state) => {
      this._isAuthServiceReady = state
    },
    error => {
      console.log("auth state error " + error);
    });
}

signIn(): void {
  this.authService.signIn(GoogleLoginProvider.PROVIDER_ID)
    .catch(reason => {
      // This is an error generated by the SocialAuthService client
      console.log("caught exception " + reason);
    });
}
```

An error is logged to the console, as expected, from angularx-social-login.js:217 (version 3.5.7). But it's not logged by my code, and neither the catch during signIn nor the initState observable error are being triggered. Is there some other way to retrieve this kind of error?

**Comment by jaibatrik** [comment, marked as answer]
`SocialAuthServiceConfig` has an `onError` field. Can you try supplying a callback to that?

**Reply by brianheilig** [reply]
Yes, onError receives this error. Thank you.
